internal/runners/publish/publish.go

```diff
@@ -78,6 +78,8 @@ type ParentIngredient struct {
 	Dependencies        []inventory_models.Dependency `json:"dependencies"`
 }
 
+var nameRegexp = regexp.MustCompile(`\w+([_-]\w+)*`)
```

Comment:
This feels too aggressive, it means that if I upload `hello-world.zip` it will default to the name `hello`.

Reply:
Actually, it works as you'd expect:

```
>build\state.exe publish c:\Users\me\tmp\hello-world.zip
█ Publish Ingredient (Beta)

Beta Feature: This feature is still in beta and may be unstable.

Publish following ingredient?
name: hello-world
namespace: org/ActiveState
version: 0.0.1
description: Not Provided

 (Y/n)
```

Reply:
Oh I see, you're using the full match, so anything not covered by this regex is considered superfluous. I thought the group was what you were stripping.